opencti-platform/opencti-graphql/src/utils/sorting.ts

```diff
@@ -0,0 +1,32 @@
+import { getAttributeMappingFromPath, isDateNumericOrBooleanAttribute, schemaAttributesDefinition } from '../schema/schema-attributes';
+import { UnsupportedError } from '../config/errors';
+
+export const buildElasticSortingForAttributeCriteria = (orderCriteria: string, orderMode: 'asc' | 'desc') => {
+  const definition = schemaAttributesDefinition.getAttributeByName(orderCriteria);
+
+  // criteria not in schema, attempt keyword sorting as a last resort
+  if (!definition) {
+    return { [`${orderCriteria}.keyword`]: { order: orderMode, missing: '_last' } };
+  }
+
+  if (isDateNumericOrBooleanAttribute(orderCriteria)) {
+    // sorting on null dates results to an error, one way to fix it is to use missing: 0
+    // see https://github.com/elastic/elasticsearch/issues/81960
+    return { [orderCriteria]: { order: orderMode, missing: definition.type === 'date' ? 0 : '_last' } };
+  }
+
+  // for sorting by object attribute, we need the sortBy def to know which internal mapping to use
+  if (definition.type === 'object' && definition.format === 'standard') {
+    const { sortBy } = definition;
+    if (sortBy) {
+      const sortByMapping = getAttributeMappingFromPath(sortBy.path);
```

Comment:
This function recursively checks the schema to find the correct mapping corresponding to the path.
sortBy path thus can be complex like `object_attr.some.inner.field`. `buildElasticSortingForAttributeCriteria` will build the right query sorting option for elastic.